server/src/main/java/org/eclipse/lsp/cobol/service/delegates/completions/KeywordCompletion.java

```diff
@@ -36,27 +39,24 @@
     this.keywords = keywords;
   }
 
-  @NonNull
   @Override
-  public Collection<String> getCompletionSource(CobolDocumentModel document) {
-    return keywords.getLabels();
+  public @NonNull Collection<CompletionItem> getCompletionItems(
+      @NonNull String token, @Nullable CobolDocumentModel document) {
+    return keywords.getLabels().stream()
+        .filter(CompletionUtils.startsWithIgnoreCase(token))
+        .map(toKeywordCompletion())
+        .collect(toList());
   }
 
-  @Nullable
-  @Override
-  public String tryResolve(@NonNull String label) {
-    return keywords.getInformationFor(label);
-  }
-
-  @NonNull
-  @Override
-  public String getSortOrderPrefix() {
-    return KEYWORDS.prefix;
-  }
-
-  @NonNull
-  @Override
-  public CompletionItemKind getKind() {
-    return CompletionItemKind.Keyword;
+  private Function<String, CompletionItem> toKeywordCompletion() {
+    return it -> {
+      CompletionItem item = new CompletionItem(it);
```

Comment:
Optional: I guess, is it possible to extract this as a  method, then one "return" can removed (implicit lambda return)